### PR TITLE
imp: don't send Content-length header for GET requests

### PR DESCRIPTION
**Pull request by timyates**
See https://github.com/micronaut-projects/micronaut-core/pull/10211

Disabled for the JDK client tests as this is broken in the JDK for 17

[The issue shows it as fixed in 19](https://bugs.openjdk.org/browse/JDK-8283544)

### CODE REVIEW

File: http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    void tearDown() {
+        if (server != null) {
+            server.stop(0);
+        }
+        applicationContext.stop();
+    }
+
+    @ParameterizedTest(name = "blocking={0}")
+    @ValueSource(booleans = {true, false})
+    @Disabled
```
Comment (yawkat) [inline http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java]: why disabled?

File: http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    void postContainsHeader(boolean blocking) {
+        MutableHttpRequest<String> post = HttpRequest.POST(PATH, "tim");
+        String retrieve = blocking
+            ? httpClient.toBlocking().retrieve(post)
+            : Flux.from(httpClient.retrieve(post)).blockFirst();
+        assertEquals("POST:3", retrieve);
+    }
+
+    @ParameterizedTest(name = "blocking={0}")
+    @ValueSource(booleans = {true})
```
Comment (yawkat) [inline http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java]: why only blocking?

**Review by yawkat — APPROVED** [review]

File: http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java (hunk 1 above)
Comment (timyates) [reply http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java]: I forgot to put it back after debugging the JDK client 😬 

Good catch! 👍

File: http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java (hunk 2 above)
Comment (timyates) [reply http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java]: I forgot to put it back after debugging the JDK client 😬 

Good catch! 👍

**Comment by timyates** [comment]
This will fail the tests here until https://github.com/micronaut-projects/micronaut-core/pull/10211 is merged

File: test-suite-http-client-tck-jdk/src/test/java/io/micronaut/http/client/tck/jdk/tests/JdkHttpMethodTests.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
 @SuppressWarnings("java:S2187") // This runs a suite of tests, but has no tests of its own
 @ExcludeClassNamePatterns({
     "io.micronaut.http.client.tck.tests.ContinueTest", // Unsupported body type errors
+    "io.micronaut.http.client.tck.tests.ContentLengthHeaderTest", // GET requests add content-length header until https://github.com/openjdk/jdk/commit/6a7709320d28d8e1593b113fdf39ab583fca3687
```
Comment (sdelamo) [inline test-suite-http-client-tck-jdk/src/test/java/io/micronaut/http/client/tck/jdk/tests/JdkHttpMethodTests.java]: can we annotate the test with [@EnabledOnJre](https://junit.org/junit5/docs/5.2.0/api/org/junit/jupiter/api/condition/EnabledOnJre.html) or similar and run it in Java 21?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    @ValueSource(booleans = {true, false})
+    void postContainsHeader(boolean blocking) {
+        MutableHttpRequest<String> post = HttpRequest.POST(PATH, "tim");
+        String retrieve = blocking
+            ? httpClient.toBlocking().retrieve(post)
+            : Flux.from(httpClient.retrieve(post)).blockFirst();
+        assertEquals("POST:3", retrieve);
+    }
+
+    @ParameterizedTest(name = "blocking={0}")
```
Comment (sdelamo) [inline http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java:78]: Change this test to verify the current behavior.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: test-suite-http-client-tck-jdk/src/test/java/io/micronaut/http/client/tck/jdk/tests/JdkHttpMethodTests.java (hunk 3 above)
Comment (timyates) [reply test-suite-http-client-tck-jdk/src/test/java/io/micronaut/http/client/tck/jdk/tests/JdkHttpMethodTests.java]: Done in https://github.com/micronaut-projects/micronaut-core/pull/10218/commits/5ea08a89dcb77a93aedcfa7d238b113143438ea6

File: http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java (hunk 4 above)
Comment (timyates) [reply http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContentLengthHeaderTest.java:78]: Done in https://github.com/micronaut-projects/micronaut-core/pull/10218/commits/5ea08a89dcb77a93aedcfa7d238b113143438ea6

**Review by sdelamo — COMMENTED** [review]
it fails for me with JDK Client also with 21. 
```
java --version
openjdk 21.0.1 2023-10-17 LTS
OpenJDK Runtime Environment Corretto-21.0.1.12.1 (build 21.0.1+12-LTS)
OpenJDK 64-Bit Server VM Corretto-21.0.1.12.1 (build 21.0.1+12-LTS, mixed mode, sharing)
```

**Comment by timyates** [comment]
Yeah...it doesn't seem fixed in the Java 21 JDK 🤔

**Comment by timyates** [comment]
I know why...

**Comment by timyates** [comment]
That should do it

**Review by sdelamo — APPROVED** [review]

**Pull request merged**